packages/block-editor/src/components/provider/index.js

```diff
@@ -0,0 +1,64 @@
+/**
+ * WordPress Dependencies
+ */
+import { Component } from '@wordpress/element';
+import { DropZoneProvider, SlotFillProvider } from '@wordpress/components';
+import { withDispatch, withSelect } from '@wordpress/data';
+import { compose } from '@wordpress/compose';
+
+class BlockEditorProvider extends Component {
+	constructor( props ) {
+		super( ...arguments );
+		props.updateEditorSettings( props.settings );
+		props.initBlocks( props.value );
+		this.persistedValue = props.select( 'core/block-editor' ).getBlocks();
+	}
+
+	componentDidUpdate( prevProps ) {
+		if ( this.props.settings !== prevProps.settings ) {
+			this.props.updateEditorSettings( this.props.settings );
+		}
+
+		if (
+			this.props.blocks !== prevProps.blocks &&
+			this.props.blocks !== this.persistedValue
+		) {
+			this.persistedValue = this.props.blocks;
```

Comment:
Noting that our current implementation of reusable blocks has the unfortunate consequence of being the root cause for an `onChange` being called here on the first selection of a block, due to how it transpires down through autocompletion prefetching triggering a `RECEIVE_BLOCKS`. It exists today as well, but was particularly unexpected for me in considering when I'd expect `onChange` to occur from a `BlockEditor` component.